packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -433,6 +467,165 @@ export class GraphQLApi extends Construct {
     });
   }
 
+  /**
+   * Adds an IAM policy statement associated with this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param resources The set of resources to allow (i.e. ...:[region]:[accountId]:apis/GraphQLId/...)
+   * @param actions The actions that should be granted to the principal (i.e. appsync:graphql )
+   */
+  public grant(grantee: IGrantable, resources: IamResources[], ...actions: string[]): Grant {
+    return Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [...resources.map(resource => Lazy.stringValue({
+        produce: () => {
+          return this.generateResourceArn(resource);
+        },
+      }))],
+      scope: this,
+    });
+
+  }
+
+  /**
+   * Adds an IAM policy statement for full access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   */
+  public grantFullAccess(grantee: IGrantable): Grant {
+    return Grant.addToPrincipal({
+      grantee,
+      actions: ['appsync:graphql'],
+      resourceArns: [
+        this.generateResourceArn({custom: '*'}),
+      ],
+      scope: this,
+    });
+  }
+
+  /**
+   * Adds an IAM policy statement for Mutation access to this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param fields The fields to grant access to that are Mutations (leave blank for all)
+   */
+  public grantMutation(grantee: IGrantable, fields?: [string]): Grant {
```

Comment:
We use the notation `string[]` traditionally.

Same comment in many places below.